WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift

```diff
@@ -1882,6 +1874,66 @@ extension ReaderStreamViewController: ReaderPostUndoCellDelegate {
 }
 
 
+// MARK: - Select Interests Display
+private extension ReaderStreamViewController {
+    func displaySelectInterestsIfNeeded(_ content: ReaderContent) {
+        guard
+            FeatureFlag.readerImprovementsPhase2.enabled,
+            content.topicType == .discover
+        else {
+            // Hide the view if we're not on the discover tab, and it exists
+            self.selectInterestsViewController?.remove()
+            return
+        }
+
+        if self.selectInterestsViewController != nil {
+            self.showSelectInterestsViewIfNeeded()
+            return
+        }
+
+        // If we're not showing the select interests view, check to see if we should
+        interestsCoordinator.shouldDisplay { [unowned self] shouldDisplay in
+            if shouldDisplay {
+                self.makeSelectInterestsViewControllerIfNeeded()
+                self.showSelectInterestsViewIfNeeded()
+            }
+        }
+    }
+
+    func showSelectInterestsViewIfNeeded() {
+        guard let controller = self.selectInterestsViewController else {
+            return
+        }
+
+        UIView.animate(withDuration: 0) {
```

Comment:
Why are we doing animation with 0 duration ?

Reply:
I tested here to discover:

1) With 0 duration: https://d.pr/v/QWiGIG
2) Without: https://d.pr/v/jQGbip

You can see that without 0 duration the screen quickly blinks. I added `withDuration: 0` back plus added a comment to explain why it's needed in 77fd1ff4374fc05b1309e0b08a21bf0479dd4fef

Reply:
good call